test/test_typed_array.cpp

```diff
@@ -0,0 +1,290 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or mplied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <compare>
+#include <cstdint>
+#include <iostream>  // For Doctest
+#include <string>
+
+#include "sparrow/typed_array.hpp"
+
+#include "array_data_creation.hpp"
+#include "doctest/doctest.h"
+
+using namespace sparrow;
+
+namespace
+{
+    template <typename O, std::integral I>
+    constexpr O to_value_type(I i)
+    {
+        if constexpr (std::is_same_v<O, numeric::float16_t>)
+        {
+            return static_cast<float>(i);
+        }
+        else if constexpr (std::is_arithmetic_v<O>)
+        {
+            return static_cast<O>(i);
+        }
+        else if constexpr (std::is_same_v<O, std::string>)
+        {
+            return std::to_string(i);
+        }
+    }
+}
+
+TEST_SUITE("typed_array")
+{
+    TEST_CASE_TEMPLATE_DEFINE("scalar", T, scalar)
+    {
+        // Element access
+
+        SUBCASE("constructor with parameter")
+        {
+            const auto array_data = make_test_array_data<T>(10, 1);
+            const typed_array<T> typed_array(array_data);
+            CHECK_EQ(typed_array.size(), 9);
+        }
+
+        SUBCASE("at")
+        {
+            const auto array_data = make_test_array_data<T>(10, 1);
+            typed_array<T> typed_array(array_data);
+            for (typename ::typed_array<T>::size_type i = 0; i < typed_array.size() - 1; ++i)
```

Comment:
why does `::typed_array`  the `::`  is necessary here and in other tests below?

Reply:
The issue comes from the declaration of the variable which have the same name as the type. I will change that